### ISSUE TITLE
Support core split overriding split functions in column space

### ISSUE THREAD

**Issue by gmbecker** [insightsengineering/rtables#785]
This will involve refactoring how how, and more importantly where, column subset expressions are generated. 

@Melkiades @edelarua I'm not able to assign labels currently but please create "gabe" and "sme_review" labels (or similar) and apply them here.  Also apply whatever I called the "collaborator request" type label.

I know how to do this and will implement it.

**Comment by shajoezhu** [comment]
Hi @gmbecker , can you please help me to better understand the rationale for this request? Thanks

**Comment by gmbecker** [comment]
@shajoezhu Certain complex tables apparently require columns along the lines of "patients enrolled and then so and so happened later", which is a subset of "patients enrolled". And these subsets can be overlapping, or interacting in other ways.

As I understand it, its something along the lines of:

```

         All Patients              Pats With A      Pats w/ A and B       Pats w/ B and not A 
----------------------------------------------------------------------------------------------
n            75                        50                 30                      15
mean                                            ...
```

etc etc. 

We have something that fully supports this in row space, but it currently isn't supported in column space because of differences in how the tabulation machinery handles row and column subsetting (the former is done by subsetting the actual dataframe, the latter is carried around as unevaluated expressions).

I have something working locally in my fork that is passing all the direct rtables tests including a new one for this functionality.

**Comment by Melkiades** [comment]
But if you do `split_cols_by` and then add a split function that does this?

**Comment by gmbecker** [comment]
@Melkiades it depends on the split function. If you use the provided combo functionality (e.g., add_combo_facet), it will work, but general custom splitting functions, specifically ones that override core splitting, are disallowed in column space. 

You can see that from the current tests:

https://github.com/insightsengineering/rtables/blob/4a445e72e303a403b2ac7ab5585ce14f211b2f7a/tests/testthat/test-split_funs.R#L353-L358

That is what I have fixed locally by changing how column subsetting expressions are handled. I'll create a PR, probably today, based on my fork.

**Comment by gmbecker** [comment]
Sorry for the delay, this got buried in other things and then lost over the thanksgiving break in america. Anyway, PR is up now

**Comment by gmbecker** [comment]
This was closed by #875
